WordPress/gutenberg
# Patterns: fix capabilities settings for pattern categories

## What?
Fixes the capabilities settings for pattern categories so Authors and below can view existing user categories in the editor inserter patterns tab. Also prevents Authors from adding new categories, but still provides the ability for them to assign existing categories to their own patterns.

## Why?
Currently Authors and contributors get API errors when trying to view pattern categories, and Authors are able to add as well as assign pattern categories

## How?

- Sets the context of the categories API call to `view` so contributors can see the user categories list
- Limits users to a select list of existing categories if they do not have create capabilities for `wp_pattern_category` 

## Testing Instructions

- As an admin go to /`wp-admin/edit-tags.php?taxonomy=wp_pattern_category` and remove any categories
- Add some new patterns with some new custom categories, not the core pattern categories
- Log in as a Contributer user and make sure the new categories appear in the editor inserter Patterns tab
- Check that going to `/wp-admin/edit-tags.php?taxonomy=wp_pattern_category` gives an insufficient permissions error.
- Log in as an Author and make sure the new categories appear in the editor inserter Patterns tab
- Add a new pattern and make sure you can only select from the custom categories added above and that you can't add any new categories, and check pattern saves correctly and is available in the assigned categories
- Go to `wp-admin/edit.php?post_type=wp_block` and choose a pattern to edit, and make sure you can only select and not create categories in the right panel
- Check that going to `/wp-admin/edit-tags.php?taxonomy=wp_pattern_category` gives an insufficient permissions error
- As an admin delete all the pattern categories again
- Login as Author and add a new pattern and check that the category option is not shown because there are now no categories to select from
- Login as an Editor and add a new pattern and make sure you can assign new categories to a pattern as well as select existing ones

## Screenshots or screencast 

Before:

https://github.com/WordPress/gutenberg/assets/3629020/aa409910-21ad-4980-9877-6f900a6f2724

After:

https://github.com/WordPress/gutenberg/assets/3629020/f941a842-f7be-40e2-a919-0c6873013888
